openedx_external_enrollments/tests/external_enrollments/tests_icc_external_enrollment.py

```diff
@@ -66,3 +73,248 @@ def test_str(self):
+                            </KEY>
+                            <KEY name="lang">
+                                <VALUE>en</VALUE>
+                            </KEY>
+                            <KEY name="calendartype">
+                                <VALUE null="null"/>
+                            </KEY>
+                            <KEY name="theme">
+                                <VALUE></VALUE>
+                            </KEY>
+                            <KEY name="timezone">
+                                <VALUE>99</VALUE>
+                            </KEY>
+                            <KEY name="mailformat">
+                                <VALUE>1</VALUE>
+                            </KEY>
+                            <KEY name="description">
+                                <VALUE null="null"/>
+                            </KEY>
+                            <KEY name="descriptionformat">
+                                <VALUE null="null"/>
+                            </KEY>
+                            <KEY name="city">
+                                <VALUE null="null"/>
+                            </KEY>
+                            <KEY name="url">
+                                <VALUE null="null"/>
+                            </KEY>
+                            <KEY name="country">
+                                <VALUE>AU</VALUE>
+                            </KEY>
+                            <KEY name="profileimageurlsmall">
+                                <VALUE>https://secure.gravatar.com/avatar/0e7b4c51e402f2d669dd0145c92025fc?s=35&amp;d=mm</VALUE>
+                            </KEY>
+                            <KEY name="profileimageurl">
+                                <VALUE>https://secure.gravatar.com/avatar/0e7b4c51e402f2d669dd0145c92025fc?s=100&amp;d=mm</VALUE>
+                            </KEY>
+                            <KEY name="customfields">
+                                <MULTIPLE></MULTIPLE>
+                            </KEY>
+                            <KEY name="preferences">
+                                <MULTIPLE></MULTIPLE>
+                            </KEY>
+                        </SINGLE>
+                    </MULTIPLE>
+                </KEY>
+                <KEY name="warnings">
+                    <MULTIPLE></MULTIPLE>
+                </KEY>
+            </SINGLE>
+        </RESPONSE>'''
+        response = Mock()
+        response.content = found_user_xml_response
+        expected_icc_user = {
+            'id': '1002',
+            'username': 'michael'
+        }
+
+        self.assertEqual(
+            self.base._get_icc_user_from_xml_response(response, 'get_user'),  # pylint: disable=protected-access
+            expected_icc_user
+        )
+
+    @patch.object(ICCExternalEnrollment, '_create_icc_user')
+    def test_validate_icc_user(self, create_icc_user_mock):
+        """
+        Testing _validate_icc_user method.
```

Comment:
Are you testing _validate_icc_user or _create_icc_user? @ivanvgh

Reply:
In this case 'validate_icc_user'  uses create_icc_user insede of it so when it is not valid it creates an icc user that is why I have to mock that method. In the other hand there is also another test case for create_icc_user.